templates/crossplane/pkg/crd_manager.go.tpl

```diff
@@ -4,151 +4,138 @@ package {{ .CRD.Names.Snake }}
-	if r.ko == nil {
-		// Should never happen... if it does, it's buggy code.
-		panic("resource manager's ReadOne() method received resource with nil CR object")
+func (c *connector) Connect(ctx context.Context, mg cpresource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*{{ .APIVersion}}.{{ .CRD.Names.Camel }})
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
 	}
-	observed, err := rm.sdkFind(ctx, r)
+	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
 	if err != nil {
 		return nil, err
 	}
-	return observed, nil
+	sess, err := session.NewSession(cfg)
+  return &external{client: svcapi.New(sess), kube: c.kube}, errors.Wrap(err, errCreateSession)
 }
 
-// Create attempts to create the supplied AWSResource in the backend AWS
-// service API, returning an AWSResource representing the newly-created
-// resource
-func (rm *resourceManager) Create(
-	ctx context.Context,
-	res acktypes.AWSResource,
-) (acktypes.AWSResource, error) {
-	r := rm.concreteResource(res)
-	if r.ko == nil {
-		// Should never happen... if it does, it's buggy code.
-		panic("resource manager's Create() method received resource with nil CR object")
-	}
-	created, err := rm.sdkCreate(ctx, r)
-	if err != nil {
-		return nil, err
-	}
-	return created, nil
+type external struct {
+	kube   client.Client
+	client svcsdkapi.{{ .SDKAPIInterfaceTypeName }}API
 }
 
-// Update attempts to mutate the supplied AWSResource in the backend AWS
-// service API, returning an AWSResource representing the newly-mutated
-// resource. Note that implementers should NOT check to see if the latest
-// observed resource differs from the supplied desired state. The higher-level
-// reonciler determines whether or not the desired differs from the latest
-// observed and decides whether to call the resource manager's Update method
-func (rm *resourceManager) Update(
-	ctx context.Context,
-	res acktypes.AWSResource,
-) (acktypes.AWSResource, error) {
-	r := rm.concreteResource(res)
-	if r.ko == nil {
-		// Should never happen... if it does, it's buggy code.
-		panic("resource manager's Update() method received resource with nil CR object")
+func (e *external) Observe(ctx context.Context, mg cpresource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*{{ .APIVersion}}.{{ .CRD.Names.Camel }})
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
-	updated, err := rm.sdkUpdate(ctx, r)
+
+	cr.SetConditions(runtimev1alpha1.Available())
+  {{- if .CRD.Ops.ReadMany }}
+	input := Generate{{ .CRD.Ops.ReadMany.InputRef.Shape.ShapeName }}(cr)
+	// TODO(muvaf): Generated conversion code has logic about the input, like if(len...)
+	// and has a return statement we can't control here in an arbitrary function.
+	// TODO(muvaf): Generated code has an assumption about the module name of the type (svcapitypes)
+	// but that doesn't always hold true.
+{{ $setCode := GoCodeSetReadManyOutput .CRD "resp" "cr" 1 }}
```

Comment:
`ACKResourceMetadata` field is not generated for Crossplane but we can certainly add something similar to show that all status fields have ARN embedded, which should already be the case for most of the AWS resources.